modules/core/02-client/legacy/v100/store_test.go

```diff
@@ -185,4 +185,21 @@ func (suite *LegacyTestSuite) TestMigrateStoreTendermint() {
 		expectedConsKey := ibctmtypes.GetIterationKey(clientStore, pruneHeight)
 		suite.Require().Nil(expectedConsKey, i)
 	}
+
+	// ensure metadata is set for unexpired consensus state
+	height := path.EndpointA.GetClientState().GetLatestHeight()
+	consState, ok := path.EndpointA.Chain.GetConsensusState(path.EndpointA.ClientID, height)
+	suite.Require().True(ok)
+	suite.Require().NotNil(consState)
+
+	processedTime, ok := ibctmtypes.GetProcessedTime(clientStore, height)
+	suite.Require().True(ok)
+	suite.Require().NotEqual(uint64(0), processedTime)
+
+	processedHeight, ok := ibctmtypes.GetProcessedHeight(clientStore, height)
+	suite.Require().True(ok)
+	suite.Require().Equal(types.GetSelfHeight(path.EndpointA.Chain.GetContext()), processedHeight)
+
+	consKey := ibctmtypes.GetIterationKey(clientStore, height)
```

Comment:
Yea, the purpose of this to ensure the unexpired consensus state metadata did not get pruned. I wrote this before adding the `addConsensusMetadata` function

Reply:
found a bug, since I was using the iteration keys for iterating, it wasn't updating the metadata since the iteration keys hadn't been set

changed the structure, set new metadata for all consensus states (using traditional iteration), then prune expired consensus states